pandas/tests/arrays/categorical/test_replace.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+
+import pandas as pd
+import pandas._testing as tm
+
+
+@pytest.mark.parametrize(
+    "to_replace,value,expected,check_types",
+    [
+        # one-to-one
+        (1, 2, [2, 2, 3], True),
+        (1, 4, [4, 2, 3], True),
+        (4, 1, [1, 2, 3], True),
+        (5, 6, [1, 2, 3], True),
```

Comment:
Does this need testing with different dtypes? e.g. with `to_replace=1`, `value="4"`

Reply:
It's doable, with limited dtype-checking as sorting the category index causes a crash on comparison on mixed dtypes